src/translators/crate.py

```diff
@@ -122,41 +123,35 @@ def get_health(self):
 
         return health
 
-    def _preprocess_values(self, e, original_attrs, col_names, fiware_servicepath):
-        values = []
-        for cn in col_names:
-            if cn == 'entity_type':
-                values.append(e['type'])
-            elif cn == 'entity_id':
-                values.append(e['id'])
-            elif cn == self.TIME_INDEX_NAME:
-                values.append(e[self.TIME_INDEX_NAME])
-            elif cn == FIWARE_SERVICEPATH:
-                values.append(fiware_servicepath or '')
-            else:
-                # Normal attributes
-                try:
-                    attr = original_attrs[cn][0]
-                    value = e[attr].get('value', None)
-                    if 'type' in e[attr] and e[attr]['type'] == 'geo:point':
-                        lat, lon = e[attr]['value'].split(',')
-                        value = [float(lon), float(lat)]
-                    elif 'type' in e[attr] and e[attr]['type'] == 'Property' \
-                         and 'value' in e[attr] \
-                         and isinstance(e[attr]['value'], dict) \
-                         and '@type' in e[attr]['value'] \
-                         and e[attr]['value']['@type'] == 'DateTime':
-                        value = e[attr]['value']['@value']
-                    elif 'type' in e[attr] and e[attr]['type'] == 'Relationship':
-                        value = e[attr].get('value', None) or \
-                                e[attr].get('object', None)
-
-                    values.append(value)
-                except KeyError:
-                    # this entity update does not have a value for the column
-                    # so use None which will be inserted as NULL to the db.
-                    values.append(None)
-        return values
+    @staticmethod
+    def _ngsi_geojson_to_db(attr):
+        return attr['value']
+
+    @staticmethod
+    def _ngsi_slf_to_db(attr):
+        if attr['type'] == 'geo:point':
+            lat, lon = attr['value'].split(',')
+            return [float(lon), float(lat)]
+
+    @staticmethod
+    def _ngsi_structured_to_db(attr):
+        attr_v = attr.get('value', None)
+        if isinstance(attr_v, dict):
+            return attr_v
+        logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
```

Comment:
Is this going to work in all cases? what if `attr` has no `type` attribute in it?